python/cugraph/cugraph/tests/test_node2vec.py

```diff
@@ -172,14 +96,202 @@ def test_node2vec_invalid(
-                                      use_padding=use_padding, p=p, q=q)
+                                      compress_result=compress, p=p, q=q)
     # Tests for invalid p
     for bad_p in invalid_pqs:
         with pytest.raises(ValueError):
             df, seeds = calc_node2vec(G, start_vertices, max_depth=max_depth,
-                                      use_padding=use_padding, p=bad_p, q=q)
+                                      compress_result=compress, p=bad_p, q=q)
     # Tests for invalid q
     for bad_q in invalid_pqs:
         with pytest.raises(ValueError):
             df, seeds = calc_node2vec(G, start_vertices, max_depth=max_depth,
-                                      use_padding=use_padding, p=p, q=bad_q)
+                                      compress_result=compress, p=p, q=bad_q)
+
+
+# Remove once bug is resolved
+@pytest.mark.parametrize(*_get_param_args("graph_file", [LINE]))
+@pytest.mark.parametrize(*_get_param_args("directed", DIRECTED_GRAPH_OPTIONS))
+def test_node2vec_line(graph_file, directed):
+    G = utils.generate_cugraph_graph_from_file(graph_file, directed=directed,
+                                               edgevals=True)
+    max_depth = 3
+    start_vertices = [0, 3, 6]
+    df, seeds = calc_node2vec(
+        G,
+        start_vertices,
+        max_depth,
+        compress_result=True,
+        p=0.8,
+        q=0.5
+    )
+
+
+@pytest.mark.parametrize(*_get_param_args("graph_file", utils.DATASETS_SMALL))
+@pytest.mark.parametrize(*_get_param_args("directed", DIRECTED_GRAPH_OPTIONS))
+@pytest.mark.parametrize(*_get_param_args("compress", COMPRESSED))
+def test_node2vec_new(
+    graph_file,
+    directed,
+    compress,
+):
+    cu_M = utils.read_csv_file(graph_file)
+
+    G = cugraph.Graph(directed=directed)
+
+    G.from_cudf_edgelist(cu_M, source="0", destination="1", edge_attr="2",
+                         renumber=False)
+    num_verts = G.number_of_vertices()
+    k = random.randint(6, 12)
+    start_vertices = random.sample(range(num_verts), k)
+    max_depth = 5
+    df, seeds = calc_node2vec(
+        G,
+        start_vertices,
+        max_depth,
+        compress_result=compress,
+        p=0.8,
+        q=0.5
+    )
+    vertex_paths, edge_weights, vertex_path_sizes = df
+
+    if compress:
+        # Paths are coalesced, meaning vertex_path_sizes is nonempty. It's
+        # necessary to use in order to track starts of paths
+        assert vertex_paths.size == vertex_path_sizes.sum()
+        if directed:
+            # directed graphs may be coalesced at any point
+            assert vertex_paths.size - k == edge_weights.size
+            # This part is for checking to make sure each of the edges
+            # in all of the paths are valid and are accurate
+            idx = 0
+            for path_idx in range(vertex_path_sizes.size):
+                for j in range(vertex_path_sizes[path_idx] - 1):
```

Comment:
`j` is never used. you could probably do something like 
```
start_path_idx = 0
end_path_idx = 0
for path_idx in range(vertex_path_sizes.size):
      end_path_idx = start_path_ix + path_idx - 1
      for idx in range (start_path_idx, end_path_idx):
```
It's just a recommendation. Or substitute `j` with `_` so that we know it's not going to be used

Reply:
substituted `j` with `_`